src/deep_learning_container.py

```diff
@@ -308,14 +308,15 @@ def main():
 
     logging.basicConfig(level=logging.ERROR)
     token = None
-
-    instance_id = _retrieve_instance_id()
-    if instance_id:
-        region = _retrieve_instance_region()
-    else:
-        token = get_imdsv2_token()
+    instance_id = None
+    region = None
+    token = get_imdsv2_token()
+    if token:
```

Comment:
The function get_imdsv2_token does return token = None if response is None or status code is outside of the bound [400,600). It is redundant to check token again before retrieving instance_id and region, since we know this is possible. We can reduce the entire 6 lines (L314-L319) to just 2 lines (L315-L316).

Reply:
How the else part would be called then if token is None?

Reply:
@kenny-ezirim is saying that the default value for token is None, so this else statement is redundant. You can instead remove conditional statement and do `fn(token=token)`